pylib_3.9.7/df/ExtractGenbankRegions.py

```diff
@@ -44,36 +61,93 @@ def execute(self, request: DataFunctionRequest) -> DataFunctionResponse:
 
         region_names = list(region_names_dict.keys())
-
         output_columns = []
+        feature_sequence_ids = []
+
         for region_name in region_names:
-            feature_sequences = [_extract_region(s, feature_key, feature_qualifier, region_name) for s in
-                                 input_sequences]
-            output_column = sequences_to_column(feature_sequences, f'{input_column.name} {region_name}', False)
-            output_column.insert_nulls(input_column.missing_null_positions)
-            output_columns.append(output_column)
-        response = DataFunctionResponse(outputColumns=output_columns)
+            feature_sequences = [_extract_region(s, feature_key, feature_qualifier, region_name)
+                                 for s in input_sequences]
+            feature_sequence_ids = [i for i, seq in zip(sequence_ids_no_nulls, input_sequences) if seq]
+
+
+            if feature_qualifier:
+                # verify handling of list of lists
+                output_column = sequences_to_column([feature_sequence[0] if len(feature_sequence) != 0 else None
+                                                     for feature_sequence in feature_sequences],
+                                        f'{input_column.name} {region_name}', False)
+                output_column.insert_nulls(input_column.missing_null_positions)
+                output_columns.append(output_column)
+
+                response = DataFunctionResponse(outputColumns = output_columns)
+            else:
+                output_feature_sequences = []
+                output_feature_ids = []
+                output_feature_descr = []
+                output_feature_start = []
+                output_feature_end = []
+                for feature_seqs, sequence_id in zip(feature_sequences, feature_sequence_ids):
+                    output_feature_ids.extend([sequence_id] * len(feature_seqs))
+                    output_feature_sequences.extend(feature_seqs)
+
+                    for feature_seq in feature_seqs:
+                        qualifiers = [v for k, v in feature_seq.features[0].qualifiers.items()]
+                        output_feature_descr.append(qualifiers[0][0])
+                        output_feature_start.append(feature_seq.features[0].location.start)
+                        output_feature_end.append(feature_seq.features[0].location.end)
+                        # output_feature_start.extend([feature_seq.features[0].location.start for feature_seq in feature_seqs])
+                        # output_feature_end.extend([feature_seq.features[0].location.end for feature_seq in feature_seqs])
+
+                output_columns = [
+                                  ColumnData(name = sequence_id_column_name, dataType = DataType.STRING,
+                                             values = output_feature_ids),
+                                  sequences_to_column(output_feature_sequences, f'{input_column.name} {region_name}', False),

```

Comment:
Are you thinking of supporting GenBank output in the future?  Might be useful as I think Biopython carries over sub-features when extracting a larger feature (so you could keep liabilities and numbering when extracting an Antibody chain)

Reply:
I thought about that as a possibility.  The Features/subfeatures bit of BioPython is exactly what I used here to support extra columns in the table - name, start, end.  I had a version which would create a Genbank column just with the sequence corresponding to the region, and then giving it annotations (see screenshot)
, but I realized that this was technically not extracting a Genbank region to a sequence column, but rather just creating another Genbank region.  

<img width="248" alt="Screenshot 2024-01-14 140726" src="https://github.com/Glysade/DataFxnPylib/assets/133695693/4437d76b-80ed-4d17-9984-644ef7ecf9fd">

Reply:
There are so many things we could do like this for Genbank functionality.  I would appreciate talking about some of the ideas I've had and what would be useful and what wouldn't.

Reply:
Happy to talk, or you could do it in our group chats.